src/Microsoft.AspNetCore.Mvc/HostingApplicationNameProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Localization;
+
+namespace Microsoft.AspNetCore.Mvc
```

Comment:
Should probably squirrel this away in some more obscure namespace. This isn't a type any normal app would use.